nucliadb/nucliadb/search/predict.py

```diff
@@ -138,10 +154,47 @@ async def send_feedback(
             )
         await self.check_response(resp, expected=204)
 
+    @predict_observer.wrap({"type": "rephrase"})
+    async def rephrase_query(self, kbid: str, item: RephraseModel) -> str:
+        if self.dummy:
+            self.calls.append(item)
+            return DUMMY_REPHRASE_QUERY
+
+        if self.onprem is False:
+            # Upload the payload
+            resp = await self.session.post(
+                url=f"{self.cluster_url}{PRIVATE_PREDICT}{REPHRASE}",
+                json=item.dict(),
+                headers={"X-STF-KBID": kbid},
+            )
+        else:
+            if self.nuclia_service_account is None:
+                error = "Nuclia Service account is not defined so could not rephrase query"
+                logger.warning(error)
+                raise SendToPredictError(error)
+            # Upload the payload
+            headers = {"X-STF-NUAKEY": f"Bearer {self.nuclia_service_account}"}
+            resp = await self.session.post(
+                url=f"{self.public_url}{PUBLIC_PREDICT}{REPHRASE}",
+                json=item.dict(),
+                headers=headers,
+            )
+        await self.check_response(resp, expected=200)
+        return await resp.read()
+
     @predict_observer.wrap({"type": "chat"})
     async def chat_query(
         self, kbid: str, item: ChatModel
     ) -> Tuple[str, AsyncIterator[bytes]]:
+        if self.dummy:
+            self.calls.append(item)
+
+            async def generate():
+                for i in [b"valid ", b"answer ", b" to"]:
+                    yield i
+
+            return ("00", generate())
```

Comment:
what is this "00" for? Do we have this documented somewhere?

Reply:
Its an ID from learning

Reply:
Could you move it to a constant or something more documented? A string literal related with learning repo has no readable meaning for nucliadb